compiler/p/codegen/PPCInstruction.cpp

```diff
@@ -1571,16 +1571,43 @@ void TR::PPCControlFlowInstruction::assignRegisters(TR_RegisterKinds kindToBeAss
          {
          cg()->traceRAInstruction(cursor = generateTrg1Src1ImmInstruction(cg(), TR::InstOpCode::cmpi4, currentNode, getTargetRegister(0), getSourceRegister(0), 0, cursor));
 
-         cg()->traceRAInstruction(cursor = generateTrg1Src1Instruction   (cg(), getOpCode2Value(), currentNode, getTargetRegister(1), getSourceRegister(1), cursor));
+         if (useRegPair())
+            {
+            cg()->traceRAInstruction(cursor = generateTrg1Src1Instruction   (cg(), getOpCode2Value(), currentNode, getTargetRegister(1), getSourceRegister(1), cursor));
+            cg()->traceRAInstruction(cursor = generateTrg1Src1Instruction   (cg(), getOpCode2Value(), currentNode, getTargetRegister(2), getSourceRegister(2), cursor));
+            }
+         else
+            {
+            cg()->traceRAInstruction(cursor = generateTrg1Src1Instruction   (cg(), getOpCode2Value(), currentNode, getTargetRegister(1), getSourceRegister(1), cursor));
+            }
          cg()->traceRAInstruction(cursor = generateConditionalBranchInstruction(cg(), TR::InstOpCode::bne, currentNode, label2, getTargetRegister(0), cursor));
 
-         if (getNumSources() == 4)
+         if (useRegPair())
             {
-            cg()->traceRAInstruction(cursor = generateTrg1Src1ImmInstruction(cg(), TR::InstOpCode::cmpi4, currentNode, getTargetRegister(0), getSourceRegister(3), 0, cursor));
-            cg()->traceRAInstruction(cursor = generateConditionalBranchInstruction(cg(), TR::InstOpCode::bne, currentNode, label2, getTargetRegister(0), cursor));
+            if (getNumSources() == 6)
```

Comment:
@yanluo7 could you please remind me what is the different number of iternary sources all about (even without pairs).

Reply:
@gita-omr  This was existing logic, it seems to come from https://github.com/eclipse/omr/blob/6e1c999e6109f80c8e09abe629610fe4b27a66ea/compiler/p/codegen/ControlFlowEvaluator.cpp#L1284-L1285
It says if the result of the first child is a long value and we are on 32 bit, it needs to add the high order of the reg pair to the source register list. In the code above it says if we have the high order register added, in addition to comparing the lower order reg we must also compare the additional reg in order to fully test the condition.

Reply:
Thanks. Sorry for being picky but then it becomes quite confusing what useRegPair() means. Does it mean for the first child or for the last two. We use it for the last two and then decide on the first one based on the number of sources which is really hard to follow inside TR::PPCControlFlowInstruction::assignRegisters()

Maybe we could add two queries: useRegPairForCond() and useRegPairForResult() or somethinglike that...

Reply:
@gita-omr I agree I think using query makes for better code -- added useRegPairForCond and renamed useRegPairForResult.